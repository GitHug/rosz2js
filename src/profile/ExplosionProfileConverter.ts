import { ExplosionProfile, BSExplosionCharacteristic, TypeName } from '../types';
import AbstractProfileConverter from './AbstractProfileConverter';

class ExplosionProfileConverter extends AbstractProfileConverter<ExplosionProfile, BSExplosionCharacteristic> {
  constructor() {
    super({
      typeName: TypeName.EXPLOSION,
      name: '-',
      diceRoll: '-',
      distance: '-',
      mortalWounds: '-'
    });
  }

  protected getProperty(bsCharacteristic: BSExplosionCharacteristic): Partial<ExplosionProfile> {
    switch (bsCharacteristic.$.name) {
      case 'Dice roll':
        return { diceRoll: bsCharacteristic._ };
      case 'Distance':
        return { distance: bsCharacteristic._ };
      case 'Mortal wounds':
        return { mortalWounds: bsCharacteristic._ };
    }
  }
}

export default ExplosionProfileConverter;
