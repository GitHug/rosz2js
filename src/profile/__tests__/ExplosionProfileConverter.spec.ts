import ExplosionProfileConverter from '../ExplosionProfileConverter';
import { BSTypeName, TypeName } from '../../Types';

describe('ExplosionProfileConverter', () => {
  it('should convert explosion characteristics', () => {
    const converter = new ExplosionProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.EXPLOSION,
        name: 'Megablast'
      },
      characteristics: [
        {
          characteristic: [
            {
              $: {
                name: 'Dice roll'
              },
              _: '42'
            },
            {
              $: {
                name: 'Distance'
              },
              _: '200cm'
            },
            {
              $: {
                name: 'Mortal wounds'
              },
              _: 'A lot'
            }
          ]
        }
      ]
    });

    expect(profile).toEqual({
      typeName: TypeName.EXPLOSION,
      name: 'Megablast',
      diceRoll: '42',
      distance: '200cm',
      mortalWounds: 'A lot'
    });
    expect(profile).toMatchSnapshot();
  });

  it('should have default values', () => {
    const converter = new ExplosionProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.EXPLOSION,
        name: 'Megablast'
      },
      characteristics: []
    });

    expect(profile).toEqual({
      typeName: TypeName.EXPLOSION,
      name: 'Megablast',
      diceRoll: '-',
      distance: '-',
      mortalWounds: '-'
    });
    expect(profile).toMatchSnapshot();
  });
});
