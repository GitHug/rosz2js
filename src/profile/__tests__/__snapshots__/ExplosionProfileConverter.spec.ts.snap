// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplosionProfileConverter should convert explosion characteristics 1`] = `
Object {
  "diceRoll": "42",
  "distance": "200cm",
  "mortalWounds": "A lot",
  "name": "Megablast",
  "typeName": "Explosion",
}
`;

exports[`ExplosionProfileConverter should have default values 1`] = `
Object {
  "diceRoll": "-",
  "distance": "-",
  "mortalWounds": "-",
  "name": "Megablast",
  "typeName": "Explosion",
}
`;
